app/src/main/java/com/duckduckgo/app/trackerdetection/api/TrackerDataDownloader.kt

```diff
@@ -86,7 +86,8 @@ class TrackerDataDownloader @Inject constructor(
             val body = response.body()!!
 
             appDatabase.runInTransaction {
-                temporaryTrackingWhitelistDao.updateAll(body.lines().map { TemporaryTrackingWhitelistedDomain(it) })
+                temporaryTrackingWhitelistDao.updateAll(body.lines().takeWhile { it.isNotBlank() }
```

Comment:
Edge case and shouldn't happen, but if the received file contains an empty line inside the body, we will not save all the received domains.

e.g:
```
domain1.com
domain2.com

domain3.com
domain4.com

```

We will store only domain1 and 2, but not domain3 and 4. All 4 domains should be stored.

Reply:
One recommendation (haven't tested how it works): Instead of `takeWhile` use `filter` or `filterNot`.

Reply:
you are right! I adjusted that to use `filter` instead.
I also added a unit test for that, but I put that in a separate PR since it would introduce a new dependency to Junit5, see https://github.com/duckduckgo/Android/pull/989
Testing simple things like this is faster using plain jvm unit tests and Junit5 allows it to keep your test classes really compact :wink: